environment/src/main/java/com/sequenceiq/environment/network/service/SubnetIdProvider.java

```diff
@@ -30,9 +30,16 @@ public String provide(NetworkDto network, Tunnel tunnel, CloudPlatform cloudPlat
             LOGGER.debug("Check failed, returning null");
             return null;
         }
-        List<CloudSubnet> selectedSubnets = cloudPlatformConnectors.get(new CloudPlatformVariant(cloudPlatform.name(), cloudPlatform.name()))
+        SubnetSelectionResult subnetSelectionResult = cloudPlatformConnectors.get(new CloudPlatformVariant(cloudPlatform.name(), cloudPlatform.name()))
                 .networkConnector()
                 .selectSubnets(new ArrayList<>(network.getSubnetMetas().values()), SubnetSelectionParameters.builder().withTunnel(tunnel).build());
-        return selectedSubnets.get(0).getId();
+        CloudSubnet selectedSubnet = subnetSelectionResult.hasResult()
+                ? subnetSelectionResult.getResult().get(0)
+                : fallback(network);
+        return selectedSubnet.getId();
+    }
+
+    private CloudSubnet fallback(NetworkDto network) {
```

Comment:
add a log here please

Reply:
Yes, thanks for pointing that out! I added it.